src/Features/Core/Portable/UnifiedSuggestions/UnifiedSuggestedActions/IUnifiedSuggestedAction.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using Microsoft.CodeAnalysis.CodeActions;
+
+namespace Microsoft.CodeAnalysis.UnifiedSuggestions
+{
+    /// <summary>
+    /// Similar to ISuggestedAction, but in a location that can be used by both local Roslyn and LSP.
+    /// </summary>
+    internal interface IUnifiedSuggestedAction
```

Comment:
All these cloned types worry me - we basically building an entire type hierarchy mimic'ing the Suggested action APIs exposed by the editor, which could be a big maintenance pain point as their API evolves. Instead of an `IUnifiedSuggestedAction`, an `ISuggestedAction` public interface that is implemented by both `SuggestedAction` and `UnifiedSuggestedAction` would ensure that any changes to the interface are picked up in both sides.

Reply:
> public interface that is implemented by both SuggestedAction and UnifiedSuggestedAction would ensure that any changes to the interface are picked up in both sides.

Core problem here is that SuggestedAction is *not* our type.  It's a VS type.  So we have our common layer feeding into both VS and VSCode/LSP.  The goal here is to have that common layer be abstracted away from VS and VSCode/LSP, so taht we can do common algorithms on that common layer, and then jsut have a thin marshalling layer to VS and VSCode.

Reply:
Oh... i do see waht you mean.  We implement ISuggestedAction (a VS type) with SuggestedAction (a roslyn type).  Yes, it might be nice if we coudl unify more here.  Could you help @allisonchou out with this @mavasani as this might be a new type of refactoring for her.

Reply:
Discussed with Manish - since we don't own `ISuggestedAction`, it will probably currently be difficult to make changes to unify `SuggestedAction`/`UnifiedSuggestedAction`. However, for other cases where both the Roslyn and LSP types are internal and defined inside Roslyn, we should create an internal interface that each pair implements (for example as Manish suggests here: https://github.com/dotnet/roslyn/pull/46082#discussion_r461804553).